### ISSUE TITLE
[cmd/telemetrygen] default value `Unset` of args `status-code` is illegal

### ISSUE THREAD

**Issue by fatsheep9146** [open-telemetry/opentelemetry-collector-contrib#25906]
### Component(s)

cmd/telemetrygen

## Description

the default value `Unset` of args `status-code` is not legal.

```
Error: expected `status-code` to be one of (Unset, Error, Ok) or (0, 1, 2), got "Unset" instead
```

## Steps to Reproduce

do not specify the args `status-code` to start `telemetrygen traces`

## Expected Result

do not fail 

### Collector version

0.83.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

**Comment by mx-psi** [comment]
Was this fixed by #25889 @fatsheep9146 ?

**Comment by fatsheep9146** [comment]
> Was this fixed by #25889 @fatsheep9146 ?

I solve this by set default value to empty string, but as @codeboten said, we'd better figure out why "unset" is not valid.

**Comment by jmsnll** [comment]
I had a look into this and it comes down to this section of code: 

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/98d0d63ba77d40d4d8af1543f9c41020d1571294/cmd/telemetrygen/internal/traces/traces.go#L125-L132

It tries on line 129 to Unmarshal the status code string to JSON but it isn't valid JSON, its not unique to `Unset` and happens for both `Ok` and `Error` too.

I'll open a merge request.
